bql/semantic/expression.go

```diff
@@ -249,6 +249,40 @@ func (e *comparisonForNodeLiteral) Evaluate(r table.Row) (bool, error) {
 	}
 }
 
+// comparisonForPredicateLiteral represents the internal representation of an expression of comparison between a binding and a predicate literal.
+type comparisonForPredicateLiteral struct {
+	op OP // operation.
+
+	lB  string // left binding.
+	rPL string // right predicate literal.
+}
+
+func (e *comparisonForPredicateLiteral) Evaluate(r table.Row) (bool, error) {
+	eL, err := cellFromRow(e.lB, r)
+	if err != nil {
+		return false, fmt.Errorf("comparisonForPredicateLiteral.Evaluate failed, the call for cellFromRow(%v, %v) returned error: %v", e.lB, r, err)
+	}
+	if eL.S != nil {
+		return false, fmt.Errorf("a string binding can only be compared with a literal of type text, got literal %q instead", strings.TrimSpace(e.rPL))
+	}
+	if eL.P == nil {
+		return false, nil
+	}
+
+	csEL, err := formatCell(eL)
+	if err != nil {
+		return false, fmt.Errorf("comparisonForPredicateLiteral.Evaluate failed, the call for formatCell(%s) returned error: %v", eL, err)
+	}
+	csER := strings.TrimSpace(e.rPL)
+
+	switch e.op {
+	case EQ:
+		return reflect.DeepEqual(csEL, csER), nil
```

Comment:
If both are strings, you can just compare them with `==`. No need for the reflect.DeepEqual.

Reply:
Thank you for getting this Tati. I have just done so.